src/helpers/EtherFiOperationParameters.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EtherFiOperationParameters is UUPSUpgradeable, OwnableUpgradeable {
+    mapping(string => mapping(address => bool)) public tagAdmins;
+    mapping(string => mapping(string => string)) public tagKeyValues;
+
+    event UpdatedAdmin(string tag, address admin, bool allowed);
+    event UpdatedKeyValue(string tag, string key, string old_value, string new_value);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() public initializer {
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+    }
+
+    function updateTagAdmin(string memory tag, address admin, bool allowed) public {
+        tagAdmins[tag][admin] = allowed;
+     
+        emit UpdatedAdmin(tag, admin, allowed);
+    }
+
+    function updateTagKeyValue(string memory tag, string memory key, string memory value) public {
```

Comment:
Why is nothing permissioned. Anyone can update tag key value...

Reply:
fixed sir